### ISSUE TITLE
Error occurred when executing MaskDetailerPipe:

### ISSUE THREAD

**Issue by xueqing0622** [ltdrdata/ComfyUI-Impact-Pack#361]
Error occurred when executing MaskDetailerPipe:

index 2 is out of bounds for dimension 0 with size 1

File "F:\ComfyUI\ComfyUI\execution.py", line 153, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
File "F:\ComfyUI\ComfyUI\execution.py", line 83, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
File "F:\ComfyUI\ComfyUI\execution.py", line 76, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
File "F:\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\impact_pack.py", line 1236, in doit
DetailerForEach.do_detail(image, segs, model, clip, vae, guide_size, guide_size_for, max_size, seed+i, steps,
File "F:\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\impact_pack.py", line 228, in do_detail
enhanced_pil, cnet_pil = core.enhance_detail(cropped_image, model, clip, vae, guide_size, guide_size_for_bbox, max_size,
File "F:\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\core.py", line 273, in enhance_detail
refined_latent = detailer_hook.cycle_latent(refined_latent)
File "F:\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\core.py", line 1308, in cycle_latent
latent = self.hook2.cycle_latent(latent)
File "F:\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\core.py", line 1911, in cycle_latent
return self.unsample(latent)
File "F:\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\core.py", line 1899, in unsample
samples = Unsampler().unsampler(self.model, self.cfg, self.sampler_name, self.steps, end_at_step,
File "F:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_Noise\nodes.py", line 213, in unsampler
noise_mask = comfy.sample.prepare_mask(latent["noise_mask"], noise, device)
File "F:\ComfyUI\ComfyUI\comfy\sample.py", line 30, in prepare_mask
noise_mask = torch.nn.functional.interpolate(noise_mask.reshape((-1, 1, noise_mask.shape[-2], noise_mask.shape[-1])), size=(shape[2], shape[3]), mode="bilinear")

Everything is ok but when set switch to 2, use UnsamplerDetailerHookProvider: 
https://www.youtube.com/watch?v=KE-2Kxy9NdQ&ab_channel=Dr.Lt.Data%27sComfyUIExtension
![image](https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/114840118/a2ec6d78-a7cb-4f07-b7c4-ab122a33ee7e)

**Comment by ltdrdata** [comment]
Oh..I forgot the mentioning.. this... in the video..

![image](https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/128333288/295fc51d-3acb-4c25-8c99-0ea3299599f0)

That feature requires this.
https://github.com/BlenderNeko/ComfyUI_Noise/pull/13

**Comment by xueqing0622** [comment]
It works now, Thx!

**Comment by kakochka** [comment]
I didn't get it. Could you explain more pls. What's that PR-13, where to get it and how to apply? If this is a node, I didn't find one in Comfy. Moreover, It'll be great if you could explain in greater details Unsampler hook, it's uses and what its options mean, cause they sound super confusing. You've gone through them in the video, but I'm still in the dark(

**Comment by ltdrdata** [comment]
@kakochka 

If you're not sure, you can uninstall the existing ComfyUI Noise. Then, go to the Manager menu, select "Channel: forked," and install the version I've modified. This way, you can use it temporarily.

![image](https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/128333288/b3b4d718-7623-447f-b0ef-6a12d9aeb325)

![image](https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/128333288/109c06e5-1ec9-4b10-9edb-a1ab8f77dee4)
